src/snowflake/connector/auth/keypair.py

```diff
@@ -57,19 +54,20 @@ def __init__(
                 object that implements the `RSAPrivateKey` interface.
             lifetime_in_seconds: number of seconds the JWT token will be valid
         """
-        super().__init__()
+        max_retry_attempts = int(
+            os.getenv(
+                "JWT_CNXN_RETRY_ATTEMPTS", AuthByKeyPair.DEFAULT_JWT_RETRY_ATTEMPTS
+            )
+        )
+        super().__init__(max_retry_attempts=max_retry_attempts, **kwargs)
+
         self._private_key: bytes | RSAPrivateKey | None = private_key
         self._jwt_token = ""
         self._jwt_token_exp = 0
         self._lifetime = timedelta(
             seconds=int(os.getenv("JWT_LIFETIME_IN_SECONDS", lifetime_in_seconds))
         )
-        self._jwt_retry_attempts = int(
-            os.getenv(
-                "JWT_CNXN_RETRY_ATTEMPTS", AuthByKeyPair.DEFAULT_JWT_RETRY_ATTEMPTS
-            )
-        )
-        self._timeout = int(
+        self._jwt_timeout = int(
```

Comment:
It is supposed to be used here:

`jwt_timed_out = self._retry_ctx.remaining_time_millis(self._jwt_timeout) < 0`

But I made a typo, so it was:

`jwt_timed_out = self._retry_ctx.remaining_time_millis(jwt_timed_out) < 0`

Which wasn't caught by the test because `jwt_timed_out = False` and `False * <any integer> = 0`... Thanks for catching this!